libsoftwarecontainer/src/softwarecontainer.cpp

```diff
@@ -55,9 +54,10 @@
 
 namespace softwarecontainer {
 
-SoftwareContainer::SoftwareContainer(std::shared_ptr<Workspace> workspace, const std::string &containerID) :
+SoftwareContainer::SoftwareContainer(std::shared_ptr<Workspace> workspace, const ContainerID id) :
```

Comment:
Would be nice to have the string prefix as well here or maybe the ContainerID type should be changed to a std::string.